crates/autopilot/src/infra/blockchain/mod.rs

```diff
@@ -57,6 +57,7 @@ impl Rpc {
 #[derive(Clone)]
 pub struct Ethereum {
     web3: DynWeb3,
+    debug_api_web3: DynWeb3,
```

Comment:
I think for the reader it's more useful to know how these web3 instances are different instead of listing for what each one is supposed to be used.
```suggestion
    unbuffered_web3: DynWeb3,
```

Reply:
That was my first version, I thought that the regular web3 also could be unbuffered, so I renamed it 😄  Reverted to `unbuffered_web3`.

Reply:
Well, you are not wrong. I'd say the difference is that for the regular `web3` instance we don't care whether it's buffered or unbuffered whereas for the `unbuffered` instance we absolutely require/expect it to be unbuffered. :)